app/controllers/supplementary/PackageInformationController.scala

```diff
@@ -31,31 +35,85 @@ import views.html.supplementary.package_information
 import scala.concurrent.{ExecutionContext, Future}
 
 class PackageInformationController @Inject()(
-  appConfig: AppConfig,
-  override val messagesApi: MessagesApi,
   authenticate: AuthAction,
-  customsCacheService: CustomsCacheService
-)(implicit ec: ExecutionContext)
+  errorHandler: ErrorHandler,
+  cacheService: CustomsCacheService
+)(implicit ec: ExecutionContext, appConfig: AppConfig, override val messagesApi: MessagesApi)
     extends FrontendController with I18nSupport {
-  import forms.supplementary.PackageInformation._
+
+  val packagesMaxElements = 99
 
   def displayForm(): Action[AnyContent] = authenticate.async { implicit request =>
-    customsCacheService.fetchAndGetEntry[PackageInformation](supplementaryCacheId, formId).map {
-      case Some(data) => Ok(package_information(appConfig, form.fill(data)))
-      case _          => Ok(package_information(appConfig, form))
+    cacheService
+      .fetchAndGetEntry[Seq[PackageInformation]](supplementaryCacheId, formId)
+      .map(items => Ok(package_information(form, items.getOrElse(Seq.empty))))
+
+  }
+
+  def submitForm(): Action[AnyContent] = authenticate.async { implicit authRequest =>
+    val actionTypeOpt = authRequest.body.asFormUrlEncoded.map(FormAction.fromUrlEncoded(_))
+    cacheService
+      .fetchAndGetEntry[Seq[PackageInformation]](supplementaryCacheId, formId)
+      .flatMap { data =>
+        implicit val packagings = data.getOrElse(Seq.empty)
```

Comment:
I'm not sure about implicit data for packagings, but it's something about good practices only, so we can stay with this

Reply:
its making the packages available for all methods , if we have option there would be further retrieval of option in every method.